stdlib/string.mli

```diff
@@ -316,6 +316,18 @@ val equal: t -> t -> bool
 (** The equal function for strings.
     @since 4.03.0 *)
 
+val hash : t -> int
+(** The hash function for strings, with the same specification as
+    {!Hashtbl.hash}. Along with the type [t], this function [hash]
+    allows the module [String] to be passed as argument to the functor
+    {!HashTbl.Make}.
+    @since 4.10.0 *)
```

Comment:
This will leave people who follow the (now recommended?) pattern of `Hashtbl.Make(String)` vulnerable to the hash table collision attacks that the seeded hashing was intended to thwart.
Would it make sense to have a note about this in the docstring, explaining that if untrusted input is used as keys when inserting, a random seed should be used?
See:
- the [docstring for Hashtbl.create](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Hashtbl.html)
- [our initial patch re: this problem](https://github.com/ocaml/ocaml/issues/5572)
- more discussion in the [PR for SipHash](https://github.com/ocaml/ocaml/pull/24)

Reply:
I agree it's better that all the tests and examples use `Hashtbl.MakeSeeded`, yes - `Hashtbl.Make` should only be used when one fully understands the consequences (which, for the F* case this comes from, they do!). Documentation changes, if any, probably want to go on the `Hashtbl.Make` functor itself, though?

Reply:
I agree re: `seeded_hash` vs `hash`, but it seems a little bit out of scope for this PR to rectify this problem outside of the `String` module.

Re: documentation changes: IIRC the `Hashtbl.Make` functor does not have any documentation (the documentation resides in the outer module or in the `S` signature)?
I think if we can refer to `Hashtbl.Make` here we can also refer to the documentation about the potential problems related to using that functor (with unseeded hashes).